tests/php/Jobs/CloneImagesOrVideosTest.php

```diff
@@ -0,0 +1,458 @@
+        $oldAnnotation = ImageAnnotationTest::create(['image_id' => $oldImage->id]);
+        $oldAnnotationLabel = ImageAnnotationLabelTest::create(['annotation_id' => $oldAnnotation->id]);
+        $this->beAdmin();
+        $project->addUserId($this->admin()->id, Role::adminId());
+
+        $request = new Request(['project' => $project, 'volume' => $volume, 'clone_annotations' => true]);
+
+        Queue::fake();
+        $this->expectsEvents('volume.cloned');
+        with(new CloneImagesOrVideos($request))->handle();
+        Queue::assertPushed(ProcessNewVolumeFiles::class);
+
+        $copy = $project->volumes()->first();
+        $newImage = $copy->images()->first();
+        $newAnnotation = $newImage->annotations()->first();
+        $newAnnotationLabel = $newAnnotation->labels()->first();
+
+        $this->assertNotNull($newAnnotation);
+        $this->assertNotNull($newAnnotationLabel);
+        $this->assertNotEquals($oldAnnotation->id, $newAnnotation->id);
+        $this->assertNotEquals($oldAnnotation->image_id, $newAnnotation->image_id);
+        $this->assertEquals($newAnnotation->image_id, $newImage->id);
+        $this->assertNotEquals($oldAnnotationLabel->id, $newAnnotationLabel->id);
+        $this->assertEquals($newAnnotation->id, $newAnnotationLabel->annotation_id);
+
+        $ignore = ['id', 'image_id'];
+        $this->assertEquals(
+            $oldAnnotation->makeHidden($ignore)->toArray(),
+            $newAnnotation->makeHidden($ignore)->toArray()
+        );
+
+        $ignore = ['id', 'annotation_id'];
+        $this->assertEquals(
+            $oldAnnotationLabel->makeHidden($ignore)->toArray(),
+            $newAnnotationLabel->makeHidden($ignore)->toArray()
+        );
+
+        // second test
+
+        $volume = VolumeTest::create([
+            'media_type_id' => MediaType::imageId(),
+            'created_at' => '2022-11-09 14:37:00',
+            'updated_at' => '2022-11-09 14:37:00',
+        ])->fresh(); // Use fresh() to load even the null fields.
+        // The target project.
+        $project = ProjectTest::create();
+        $oldImage->volume_id = $volume->id;
+        $oldImage->save();
+        // there are three labels in total
+        $l2 = ImageAnnotationLabelTest::create(['annotation_id' => $oldAnnotation->id]);
+        $l3 = ImageAnnotationLabelTest::create(['annotation_id' => $oldAnnotation->id]);
+
+        $request = new Request(['project' => $project, 'volume' => $volume, 'clone_annotations' => true,
+            'only_annotation_labels' => [$l2->label_id, $l3->label_id]]);
+
+        with(new CloneImagesOrVideos($request))->handle();
+
+        $copy = $project->volumes()->first();
+        $newImage = $copy->images()->first();
+        $newAnnotation = $newImage->annotations()->first();
+        $newAnnotationLabels = $newAnnotation->labels()->get();
+
+        $this->assertEquals(2, $newAnnotationLabels->count());
```

Comment:
Please also check the `label_id` of the annotation labels (similar to image/video labels above).